airflow/providers/google/cloud/hooks/compute_engine_ssh.py

```diff
@@ -0,0 +1,211 @@
+        :type cmd: union[str, List[str]]
+        """
+        self.log.info('Executing command: %s', str(cmd))
+        process = subprocess.Popen(cmd)
+        output = process.communicate()[0]
+        returncode = process.returncode
+        if returncode:
+            # Error
+            self.log.error('Command returned error status %s', returncode)
+        if output:
+            self.log.info(output)
+
+    def create_ssh_key(self,
+                       oslogin: Any,
+                       account: str,
+                       private_key_file: Optional[str] = None,
+                       expire_time: int = 300) -> str:
+        """
+        Generate a temporary SSH key and apply it to the specified account
+
+        :param oslogin:
+        :param account:
+        :param private_key_file:
+        :param expire_time:
+        """
+        private_key_file = private_key_file or '/tmp/key-' + str(uuid.uuid4())
+        self.execute_in_compute(['ssh-keygen', '-t', 'rsa', '-N', '', '-f', private_key_file])
+
+        with open(private_key_file + '.pub', 'r') as original:
+            public_key = original.read().strip()
+
+        # Expiration time is in microseconds.
+        expiration = int((time.time() + expire_time) * 1000000)
+
+        body = {
+            'key': public_key,
+            'expirationTimeUsec': expiration,
+        }
+        oslogin.users().importSshPublicKey(parent=account, body=body).execute_in_compute()
+        return private_key_file
+
+    @GoogleBaseHook.fallback_to_default_project_id
+    def run(self,
+            cmd: str, project_id: str, instance: str, zone: str,
+            private_key_file: Optional[str] = None, expire_time: int = 300,
+            hostname: Optional[str] = None, tunnel: bool = False,
+            clean_up_private_key: bool = True):
+        """
+        Run a command on a remote system.
+
+        :param cmd:
+        :param project_id:
+        :param instance:
+        :param zone:
+        :param private_key_file:
+        :param expire_time:
+        :param hostname:
+        :param tunnel:
+        :param clean_up_private_key:
+        :return:
+        """
+
+        hostname = self.get_hostname(project_id=project_id,
+                                     instance=instance,
+                                     zone=zone,
+                                     hostname=zone)
+
+        account = self.get_service_account()
+        oslogin = self.get_oslogin()
+
+        private_key_file = self.create_ssh_key(oslogin=oslogin,
+                                               account=account,
+                                               private_key_file=private_key_file,
+                                               expire_time=expire_time)
+
+        username = self.get_username()
+
+        if tunnel and "--tunnel-through-iap" not in cmd:
+            cmd = cmd + " " + "--tunnel-through-iap"
+
+        ssh_command = [
```

Comment:
We should try to do it with paramiko. We can then use more advanced features. Although I know it will be a bit more difficult then.